### PR TITLE
New board test for pcbX and pcbY and replacements

### PR DESCRIPTION
**Pull request by Slaviiiii**

### CODE REVIEW

File: src/lib/builder/board-builder.ts [hunk 1]
```diff
@@ -78,7 +69,7 @@ export class BoardBuilderClass
   }
 
   async build(bc: Type.BuildContext): Promise<Type.AnyElement[]> {
-    const required_props = ["width", "height", "center_x", "center_y"]
+    const required_props = ["width", "height", "pcbX", "pcbY"]
```
Comment (seveibar) [inline src/lib/builder/board-builder.ts:91]: Can you add a quick remapping condition here for backwards compatibility. Eg if (center_x !== undefined) props.pcbX = center_x

File: tests/board-builder/new-board.test.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+test("[smoke] new board test", async (t) => {
+  const { pb, logSoup } = getTestFixture(t)
+
+  const soup = await pb
+    .add("board", (bb) => {
+      bb.setProps({ width: 10, height: 10, pcbX: 20, pcbY: 0 })
+    })
+    .build()
+
```
Comment (seveibar) [inline tests/board-builder/new-board.test.ts:13]: Can you use su(soup) to add a condition and check the board to make sure that the boards position is correct.

When its possible to create conditions thats preferable to prevent breakage

**Review by seveibar — CHANGES_REQUESTED** [review]

File: src/lib/builder/board-builder.ts (hunk 1 above)
Comment (Slaviiiii) [reply src/lib/builder/board-builder.ts:72]: https://github.com/tscircuit/builder/pull/76/commits/297512eb900697073bcf8bcc9f70802567bd4afb

File: tests/board-builder/new-board.test.ts (hunk 2 above)
Comment (Slaviiiii) [reply tests/board-builder/new-board.test.ts:13]: [ce8f0ce](https://github.com/tscircuit/builder/pull/76/commits/ce8f0ce0b04c04b99668cabcc3821435de5bf02f)

File: src/lib/builder/board-builder.ts (hunk 1 above)
Comment (Slaviiiii) [reply src/lib/builder/board-builder.ts:91]: https://github.com/tscircuit/builder/pull/76/commits/f8a5e64cf05b3e37efca3d2a825e353cb581ee93

Comment (seveibar) [reply src/lib/builder/board-builder.ts:91]: nice, you can use `LegacyBoardProps` as a type to fix the type issue i think

Comment (Slaviiiii) [reply src/lib/builder/board-builder.ts:91]: I do not see a type like that

Comment (seveibar) [reply src/lib/builder/board-builder.ts:91]: you'll need to create it yep

Comment (seveibar) [reply src/lib/builder/board-builder.ts:91]: just throw it up at the top of the file

Comment (Slaviiiii) [reply src/lib/builder/board-builder.ts:91]: I guess I only need to use it for the setProps prob in that exact one function nowhere else

Comment (seveibar) [reply src/lib/builder/board-builder.ts:91]: yep!

File: src/lib/builder/board-builder.ts [hunk 3]
```diff
@@ -3,13 +3,28 @@ import {
...
 import { createTraceHintBuilder } from "./trace-hint-builder"
 import { createNetBuilder } from "./net-builder/net-builder"
+import type { BoardProps as OriginalBoardProps } from "@tscircuit/props"
+
+interface ExtendedBoardProps extends OriginalBoardProps {
+  board_thickness?: number | string
+  center?: Type.Point
+  pcbX?: number | string
+  pcbY?: number | string
+}
```
Comment (seveibar) [inline src/lib/builder/board-builder.ts:20]:
```suggestion
```

Just use `BoardProps` from `@tscircuit/props`. You'll need to update it to the latest version

File: src/lib/builder/board-builder.ts [hunk 4]
```diff
@@ -3,13 +3,28 @@ import {
...
+
+export interface LegacyBoardProps extends ExtendedBoardProps {
+  center_x?: number | string
+  center_y?: number | string
+  pcbX?: number | string
+  pcbY?: number | string
+}
```
Comment (seveibar) [inline src/lib/builder/board-builder.ts:27]:
```suggestion
export interface LegacyBoardProps {
  center_x?: number | string
  center_y?: number | string
}
```

File: src/lib/builder/board-builder.ts [hunk 5]
```diff
@@ -53,56 +60,73 @@ export interface BoardBuilder {
...
 }
 
 export class BoardBuilderClass
   extends GroupBuilderClass
   implements BoardBuilder
 {
-  builder_type: "board_builder" = "board_builder"
-  props: Partial<BoardProps>
+  builder_type = "board_builder" as const
+  props: Partial<ExtendedBoardProps>
```
Comment (seveibar) [inline src/lib/builder/board-builder.ts:72]:
```suggestion
  props: Partial<BoardProps>
```

File: src/lib/builder/board-builder.ts [hunk 6]
```diff
@@ -53,56 +60,73 @@ export interface BoardBuilder {
...
     return [
       ...(await super.build(bc)),
       {
         type: "pcb_board",
         center: this.props.center
           ? bc.convert(this.props.center)
           : {
-              x: bc.convert(this.props.center_x!),
-              y: bc.convert(this.props.center_y!),
+              x: convertProp(this.props.pcbX),
```
Comment (seveibar) [inline src/lib/builder/board-builder.ts:115]: revert this change

File: src/lib/builder/board-builder.ts [hunk 7]
```diff
@@ -53,56 +60,73 @@ export interface BoardBuilder {
...
+      ? bc.convert(this.addUnitIfNeeded(this.props.board_thickness))
       : 1.2
 
+    const convertProp = (prop: string | number | undefined): number => {
+      if (prop === undefined) {
+        throw new Error("Property is undefined")
+      }
+      return bc.convert(this.addUnitIfNeeded(prop)) as number
+    }
+
```
Comment (seveibar) [inline src/lib/builder/board-builder.ts:107]:
```suggestion
```

File: src/lib/builder/board-builder.ts [hunk 8]
```diff
@@ -53,56 +60,73 @@ export interface BoardBuilder {
...
-    const required_props = ["width", "height", "center_x", "center_y"]
+    const required_props = ["width", "height", "pcbX", "pcbY"]
     for (const prop of required_props) {
       if (this.props[prop] === undefined) {
         throw new Error(`<board /> "${prop}" is not set`)
       }
     }
     bc.board_thickness = this.props.board_thickness
-      ? bc.convert(this.props.board_thickness)
+      ? bc.convert(this.addUnitIfNeeded(this.props.board_thickness))
```
Comment (seveibar) [inline src/lib/builder/board-builder.ts:98]:
```suggestion
      ? bc.convert(this.props.board_thickness)
```

File: src/lib/builder/board-builder.ts [hunk 9]
```diff
@@ -53,56 +60,73 @@ export interface BoardBuilder {
...
+  props: Partial<ExtendedBoardProps>
   declare addables: BoardBuilderAddables
 
   constructor(project_builder?: ProjectBuilder) {
     super(project_builder)
     this.props = {}
   }
 
-  setProps(props: BoardProps): this {
+  setProps(props: LegacyBoardProps): this {
```
Comment (seveibar) [inline src/lib/builder/board-builder.ts:80]:
```suggestion
  setProps(props: LegacyBoardProps & BoardProps): this {
```

**Review by seveibar — CHANGES_REQUESTED** [review]
